Source/JavaScriptCore/runtime/JSArray.cpp

```diff
@@ -939,48 +947,61 @@ JSArray* JSArray::fastToSpliced(JSGlobalObject* globalObject, CallFrame* callFra
         resultButterfly->setVectorLength(vectorLength);
         resultButterfly->setPublicLength(newLength);
 
-        auto copyArrayPrefixElements = [&]<typename T, typename U>(T* resultBuffer, U* sourceBuffer) {
-            copyArrayElements<ArrayFillMode::Empty, NeedsGCSafeOps::No>(resultBuffer, 0, sourceBuffer, 0, start, sourceType);
-        };
-        auto copyArraySuffixElements = [&]<typename T, typename U>(T* resultBuffer, U* sourceBuffer) {
-            copyArrayElements<ArrayFillMode::Empty, NeedsGCSafeOps::No>(resultBuffer, start + insertCount, sourceBuffer, start + deleteCount, length - start - deleteCount, sourceType);
-        };
-
         auto* sourceButterfly = this->butterfly();
-        if (hasDouble(resultIndexingType)) {
+        uint32_t prefixResultOffset = 0;
+        uint32_t prefixSourceOffset = 0;
+        uint32_t prefixSourceLength = start;
+        uint32_t suffixResultOffset = start + insertCount;
+        uint32_t suffixSourceOffset = start + deleteCount;
+        uint32_t suffixSourceLength = length - start - deleteCount;
```

Comment:
Are these `uint32_t` within the range of `uint32_t`? Isn't it possible that they overflow?